runtime/libffi/z/ffi64.c

```diff
@@ -764,6 +766,50 @@ ffi_prep_args (unsigned char *stack, extended_cif *ecif)
 
 }
 
+/**
+ * Helper functions to know if the given struct needed to be treated for Complex
+ * type for Float or Doubles
+ */
+unsigned short
+get_ffi_element_type_in_struct(ffi_type *arg_type)
+{
+  while ((FFI_TYPE_STRUCT == arg_type->type)
+        && (NULL != arg_type->elements[0])
+        && (NULL == arg_type->elements[1])
+  ) {
+    arg_type = arg_type->elements[0];
+  }
+  return arg_type->type;
+}
+
+unsigned short
+ffi_check_struct_for_complex(ffi_type *arg_type)
+{
+  if (FFI_TYPE_STRUCT == arg_type->type)
+  {
+    unsigned short firstArgType = get_ffi_element_type_in_struct(arg_type->elements[0]);
+    if (FFI_TYPE_FLOAT == firstArgType)
+    {
+      if ((NULL != arg_type->elements[1])
+          && (NULL == arg_type->elements[2])
+          && (FFI_TYPE_FLOAT == get_ffi_element_type_in_struct(arg_type->elements[1]))
+      ) {
+        return FFI_TYPE_STRUCT_FF;
+      }
+    }
+    else if (FFI_TYPE_DOUBLE == firstArgType)
+    {
+      if ((NULL != arg_type->elements[1])
+          && (NULL == arg_type->elements[2])
+          && (FFI_TYPE_DOUBLE == get_ffi_element_type_in_struct(arg_type->elements[1]))
+      ) {
+        return FFI_TYPE_STRUCT_DD;
+      }
+    }
+  }
+  return arg_type->type;
+}
+
 /*======================== End of Routine ============================*/
```

Comment:
The code above should be added after this line which marks the end of `ffi_prep_args()`.

Reply:
Fixed in https://github.com/eclipse-openj9/openj9/pull/19645/commits/38245ec2872c2b2664f6fdd3947f80e71e63c57d#diff-7e388e6ac3235f9470ab3d7607ffbe1c8fceacddd69ac992fdae6cda3e99257d